src/common/host-utils.js

```
@@ -0,0 +1,134 @@
+/**
+ * @fileOverview
+ * @name hosts-utils.js
+ * @author Travis Crist
+ */
+
+const fetch = require('ilp-fetch')
+const logger = require('riverpig')('codius-cli:host-utils')
+const config = require('../config.js')
+const BigNumber = require('bignumber.js')
+const sampleSize = require('lodash.samplesize')
+const { getCurrencyDetails } = require('../common/price.js')
+
+async function fetchHostPrice (host, duration, manifestJson) {
+  return new Promise((resolve, reject) => {
+    fetch(`${host}/pods?duration=${duration}`, {
+      headers: {
+        Accept: `application/codius-v${config.version.codius.min}+json`,
+        'Content-Type': 'application/json'
+      },
+      method: 'OPTIONS',
+      body: JSON.stringify(manifestJson)
+    }).then(async (res) => {
+      if (res) {
+        resolve({ host, response: await res.json() })
```

it shouldn't try to get the json if it gets a non-200 error. instead, it should return the error text